fem/fespace.cpp

```diff
@@ -615,6 +667,62 @@ FiniteElementSpace::AddDependencies(SparseMatrix& deps, Array<int>& master_dofs,
    }
 }
 
+void FiniteElementSpace
+::AddEdgeFaceDependencies(SparseMatrix &deps, Array<int> &master_dofs,
+                          const FiniteElement *master_fe,
+                          Array<int> &slave_dofs, int slave_face,
+                          const DenseMatrix *pm) const
+{
+   // In variable-order spaces in 3D, we need to only constrain interior face
+   // DOFs (this is done one level up), since edge dependencies can be more
+   // complex and are primarily handled by edge-edge dependencies. The one
+   // exception is edges of slave faces that lie in the interior of the master
+   // face, which are not covered by edge-edge relations. This function finds
+   // such edges and makes them constrained by the master face.
+   // See also https://github.com/mfem/mfem/pull/1423#issuecomment-633916643
+
+   Array<int> V, E, Eo; // TODO: LocalArray
+   mesh->GetFaceVertices(slave_face, V);
+   mesh->GetFaceEdges(slave_face, E, Eo);
+   MFEM_ASSERT(V.Size() == E.Size(), "");
+
+   DenseMatrix I;
+   IsoparametricTransformation edge_T;
+   edge_T.SetFE(&SegmentFE);
+
+   // constrain each edge of the slave face
+   for (int i = 0; i < E.Size(); i++)
+   {
+      int a = i, b = (i+1) % V.Size();
+      if (V[a] > V[b]) { std::swap(a, b); }
+
+      DenseMatrix &edge_pm = edge_T.GetPointMat();
+      edge_pm.SetSize(2, 2);
+
+      // copy two points from the face point matrix
+      double mid[2];
+      for (int j = 0; j < 2; j++)
+      {
+         edge_pm(j, 0) = (*pm)(j, a);
+         edge_pm(j, 1) = (*pm)(j, b);
+         mid[j] = 0.5*((*pm)(j, a) + (*pm)(j, b));
+      }
+
+      // check that the edge does not coincide with the master face's edge
+      const double eps = 1e-14;
+      if (mid[0] > eps && mid[0] < 1-eps &&
+          mid[1] > eps && mid[1] < 1-eps)
+      {
+         int order = GetEdgeDofs(E[i], slave_dofs, 0);
```

Comment:
I don't think you need the magic number `0`.

Reply:
This is intentional, same as above.